src/plugins/telemetry_collection_manager/server/plugin.ts

```diff
@@ -157,7 +157,10 @@ export class TelemetryCollectionManagerPlugin
     const soClient = config.unencrypted
       ? collectionSoService.getScopedClient(config.request)
       : collectionSoService.createInternalRepository();
-    return { callCluster, timestamp, usageCollection, esClient, soClient };
+    // Provide the kibanaRequest so opted-in plugins can scope their custom clients only if the request is not encrypted
+    const kibanaRequest = config.unencrypted ? request : void 0;
```

Comment:
why `void 0` instead of `undefined`?

Reply:
It's essentially the same. But JS allows you do do `let undefined = 1`, while `void 0` is univoquely `undefined` :)